src/main/java/com/minecolonies/coremod/colony/requestsystem/locations/EntityLocation.java

```diff
@@ -70,13 +69,14 @@ private void checkEntity()
     public BlockPos getInDimensionLocation()
     {
         checkEntity();
-        if (entity == null || entity.get() == null)
+        final Entity entityRef = entity.get();
```

Comment:
This really does not matter at all..........
No change needed, just wanted to inform you that the HotSpot VM does this automatically, and as such these optimisations are not needed.

Next time PRs with arbitrary changes like this will get rejected.

Reply:
ah the JVM already does that, good to know

Reply:
This is a WeakReference - which has special semantics - you need to store the contained reference in a local variable or another field otherwise the GC can come and clear the WeakReference at any time. So calling get() twice in this case can result in different outcomes.